addons/l10n_sa_edi/models/account_journal.py

```diff
@@ -308,15 +336,21 @@ def _get_node(xpath_str):
 
     # ====== Index Chain & Previous Invoice Calculation =======
 
+    def _l10n_sa_edi_create_new_chain(self):
+        self.ensure_one()
+        zatca_format = self.env.ref('l10n_sa_edi.edi_sa_zatca')
+        date = zatca_format._l10n_sa_get_zatca_datetime(datetime.now()).strftime('%Y-%m-%dT%H:%M:%SZ')
+        return self.env['ir.sequence'].create({
```

Comment:
Why not just change the existing sequence back to 1 when needed?  (maybe I am missing something)

Reply:
Hi @jco-odoo i believe the ICV should still be a unique identifier for all the documents sent to zatca
resetting it to 1 means you would have duplicates

The ICV seems to be per solution, and has no to have no-gaps

Reply:
Hi @jco-odoo i believe the ICV should still be a unique identifier for all the documents sent to zatca
resetting it to 1 means you would have duplicates

The ICV is to be per unique per onboarding, and has no to have no-gaps

https://zatca1.discourse.group/t/clarification-on-icv-and-invoice-number/2877/4
https://zatca1.discourse.group/t/icv-pih-for-invoice-credit-debit/1460/4


* I think it would technically be okay for us reset the counter to 1, but I guess we would lose the 'onboarding cycle' information